src/unstract/sdk/adapters/llm/exceptions.py

```diff
@@ -24,14 +25,29 @@ def parse_llm_err(e: Exception) -> LLMError:
     Returns:
         LLMError: Unstract's LLMError object
     """
+    # Avoid wrapping LLMError objects again
+    if isinstance(e, LLMError):
+        return e
+
     if isinstance(e, ResponseValidationError):
-        return VertexAILLM.parse_llm_err(e)
+        err = VertexAILLM.parse_llm_err(e)
     elif isinstance(e, OpenAIAPIError):
-        return OpenAILLM.parse_llm_err(e)
+        err = OpenAILLM.parse_llm_err(e)
     elif isinstance(e, AnthropicAPIError):
-        return AnthropicLLM.parse_llm_err(e)
+        err = AnthropicLLM.parse_llm_err(e)
     elif isinstance(e, MistralException):
-        return MistralLLM.parse_llm_err(e)
+        err = MistralLLM.parse_llm_err(e)
     elif isinstance(e, GoogleAPICallError):
```

Comment:
There could be cases where GoogleAPICallError is thrown for Gemini/VertexAI also.... Instead we should allow the the context (in this case, the llm instance) itself to take us to the leaf implementation of the parse()

Reply:
Answered [here](https://github.com/Zipstack/unstract-sdk/pull/120#discussion_r1869046536)